mod_form.php

```diff
@@ -325,12 +346,97 @@ private function disable_grade_rules() {
     }
 
+
+    private function add_property_selection() {
+        $mform = $this->_form;
+        $mform->addElement("hidden", "mumie_selected_task_properties", "");
+        $mform->setType("mumie_selected_task_properties", PARAM_RAW);
+        $taskproperties = array(array(get_string('mumie_form_due_date', 'mod_mumie'), "duedate"),
+                        array(get_string('mumie_form_activity_container', 'mod_mumie'), "launchcontainer"),
+                        array(get_string('mumie_form_points', 'mod_mumie'), "points"));
+                        $taskpropertytable = new \html_table();
+
+        foreach ($taskproperties as $taskproperty) {
+            $checkboxhtml = html_writer::checkbox("task_property", $taskproperty[1], false);
+            $taskpropertytable->data[] = array($taskproperty[0], $checkboxhtml);
+        }
+
+        $htmltaskpropertytable = html_writer::table($taskpropertytable);
+
+        $mform->addElement(
+            'html',
+            '<div>'
+            . get_string('mumie_form_task_properties_selection_info', 'mod_mumie')
+            .'</div>'
+            .'<div class="mumie_table">'
+            . $htmltaskpropertytable
+            . '</div>' );
+
+    }
+    private function add_task_selection() {
+        global $COURSE;
+        $cm = &$this->_cm;
+        $mform = $this->_form;
+
+        $mform->addElement("hidden", "mumie_selected_tasks", "");
+        $mform->setType("mumie_selected_tasks", PARAM_RAW);
+        $mumiemodules = get_all_instances_in_course("mumie", $COURSE);
+
+        if (!is_null($cm)) {
+            $mumiemodules = array_filter($mumiemodules, function($elem) use($cm){
+                return !($elem->id === $cm->instance);
+            });
+        }
+
+        usort($mumiemodules, function($module1, $module2) {
+            return $module1->section > $module2->section;
+        });
+
+        $section = null;
+        $tasktable = null;
+        $tasktables = array();
+
+        foreach ($mumiemodules as $mumiemodule) {
+            $modulesection = $mumiemodule->section;
+            if ($section != $modulesection) {
+                $tasktable = new \html_table();
+                array_push($tasktables, $tasktable);
+                $section = $modulesection;
+                $sectionname = get_section_name($COURSE->id, $modulesection);
+                $checkboxhtml = html_writer::checkbox("section", $modulesection, false);
+                $tasktable->head = array(
+                    $sectionname,
+                    $checkboxhtml
+                );
+            }
+            $checkboxhtml = html_writer::checkbox("task", $mumiemodule->id, false, '',
+            array(
+                "section" => $modulesection
+            ));
+            $tasktable->data[] = array($mumiemodule->name, $checkboxhtml);
+        }
+
+        $htmltasktables = "";
+        foreach ($tasktables as $a) {
+            $htmltasktables = $htmltasktables . html_writer::table($a);
+        }
```

Comment:
We can make a few minor adjustments to increase readability. As a bonus, the modules are sorted automatically without usort.

```suggestion
        $tables = array();
        foreach ($mumiemodules as $module) {
            $section = $module->section;
            if (!array_key_exists($section, $tables)) {
                $table = new \html_table;
                $table->head = array(
                    get_section_name($COURSE->id, $section),
                    html_writer::checkbox("section", $section, false)
                );
                $tables[$section] = $table;
            } else {
                $table = $tables[$section];
            }
            $checkboxhtml = html_writer::checkbox(
                "task",
                $module->id,
                false,
                '',
                array(
                    "section" => $section
            ));
            $table->data[] = array($module->name, $checkboxhtml);
        }
        
        $htmltasktables = "";
        foreach ($tables as $table) {
            $htmltasktables .= html_writer::table($table);
        }
```